turtlebot3_automatic_parking/turtlebot3_automatic_parking/turtlebot3_automatic_parking.py

```diff
@@ -290,21 +317,48 @@ def _run(self):
                     self.get_logger().info("Rotation Done.")
 
             elif self.parking_sequence == 4:
-                if yaw - self.init_yaw > -pi / 2:
-                    cmd_vel.linear.x = 0.0
-                    cmd_vel.angular.z = -0.2
+                if self.new_center[1] > 0:
+                    if self.target_yaw is None:
+                        self.target_yaw = yaw - (pi / 2)
+                    if yaw - self.target_yaw > 0.1:
+                        cmd_vel.linear.x = 0.0
+                        cmd_vel.angular.z = -0.2
+                    else:
+                        cmd_vel.linear.x = 0.0
+                        cmd_vel.angular.z = 0.0
+                        self.parking_sequence += 1
+                        self.get_logger().info("Approach spot.")
+
                 else:
-                    cmd_vel.linear.x = 0.0
-                    cmd_vel.angular.z = 0.0
-                    self.parking_sequence += 1
-                    self.get_logger().info("Approach spot.")
+                    if self.target_yaw is None:
+                        self.target_yaw = yaw + (pi / 2)
+                    if self.target_yaw - yaw > 0.1:
+                        cmd_vel.linear.x = 0.0
+                        cmd_vel.angular.z = 0.2
+                    else:
+                        cmd_vel.linear.x = 0.0
+                        cmd_vel.angular.z = 0.0
+                        self.parking_sequence += 1
+                        self.get_logger().info("Approach spot.")
 
             elif self.parking_sequence == 5:
                 ranges = []
-                for i in range(150, 210):
+                min_scan_index = int(
+                    (math.radians(150) - self.scan.angle_min) / self.scan.angle_increment
+                    )
+                max_scan_index = int(
+                    (math.radians(210) - self.scan.angle_min) / self.scan.angle_increment
+                    )
```

Comment:
We try to avoid using hardcoded numeric values as magic numbers whenever possible. It would be better to declare it as a global variable and use it that way.
```suggestion
REAR_SCAN_ANGLE_MIN = 150
REAR_SCAN_ANGLE_MAX = 210
                    (math.radians(REAR_SCAN_ANGLE_MIN) - self.scan.angle_min) / self.scan.angle_increment
                    )
                max_scan_index = int(
                    (math.radians(REAR_SCAN_ANGLE_MAX) - self.scan.angle_min) / self.scan.angle_increment
                    )
```

Reply:
Please check for magic numbers throughout the code.

Reply:
[153ade7]
Fixed